libdevice/include/asan_libdevice.hpp

```diff
@@ -63,16 +63,19 @@ struct LocalArgsInfo {
   uint64_t SizeWithRedZone = 0;
 };
 
+constexpr std::size_t ASAN_MAX_NUM_REPORTS = 10;
+
 struct LaunchInfo {
-  uintptr_t PrivateShadowOffset =
-      0; // don't move this field, we use it in AddressSanitizerPass
+  // Don't move this field, we use it in AddressSanitizerPass
+  uintptr_t PrivateShadowOffset = 0;
 
   uintptr_t LocalShadowOffset = 0;
   uintptr_t LocalShadowOffsetEnd = 0;
-  DeviceSanitizerReport SanitizerReport;
 
   uint32_t NumLocalArgs = 0;
-  LocalArgsInfo *LocalArgs = nullptr; // ordered by ArgIndex
+  LocalArgsInfo *LocalArgs = nullptr; // Ordered by ArgIndex
+
+  DeviceSanitizerReport SanitizerReport[ASAN_MAX_NUM_REPORTS];
```

Comment:
Are there any concerns about register-pressure related to this? Would it make sense to somehow let users use a more light-weight version with fewer reports?

Reply:
`LaunchInfo` is allocated in shared USM, so I think it won't cause register-pressure here.